### DISCUSSION TITLE
`@rearchWidget`?

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#25] [Q&A] [answered]
In the `Widgets` doc section you show the `@rearchWidget` decorator.
https://rearch.gsconrad.com/flutter/widgets

> Any parameters you add to @rearchWidget functions are automatically passed onto the generated widget's constructor, similar to @factoryCapsule.

Do you use a generator?

**Comment by busslina** [comment]
I'm studying the `examples` and seems you don't use a generator there.

**Reply by GregoryConrad** [reply]
Yup, they currently use the RearchConsumer, at least until I can develop the rearchWidget macro with static metaprogramming.

**Comment by busslina** [comment]
Would you say that I followed good practices here, in my first ReArch Flutter hello world?
https://github.com/busslina/rearch-flutter-test/blob/master/lib/hello_world.main.dart

**Reply by GregoryConrad** [reply]
Just checking that one file, and yes looks pretty good. Two things:
1. `Widget build()` should often use a normal function body and have a return keyword in there to help with formatting.
2. `onPressed: () => use(counterCapsule).incrementCounter(),`
should become
`onPressed: use(counterCapsule).incrementCounter`,

**Reply by GregoryConrad** [reply]
Neither are logic bugs, but will help with code style and performance respectively

**Comment by GregoryConrad** [comment, marked as answer]
The docs jump the gun a little bit in preparation for when static metaprogramming comes out. For now, you must manually extend the RearchConsumer.

**Reply by busslina** [reply]
I suppose ReArch is not compatible with Flutter Hooks (which I never used yet)

**Reply by GregoryConrad** [reply]
You could use both side by side, but it’s better to just stick with one. They aren’t interoperable 

however , anything you can do in hooks you can also do in ReArch, with about near the same syntax
